dbt/adapters/athena/impl.py

```diff
@@ -419,6 +419,10 @@ def clean_up_table(self, relation: AthenaRelation) -> None:
         if table_location := self.get_glue_table_location(relation):
             self.delete_from_s3(table_location)
 
+    @available
+    def generate_unique_temporary_table_suffix(self, suffix_initial="__dbt_tmp", length=8):
+        return f"{suffix_initial}_{str(uuid4())[:length]}"
```

Comment:
nit: I don't see much need of adding length here. I believe that `{suffix_initial}_{str(uuid4())}` is enough, as we do for s3_data_naming.